pkg/media/protectionMetadata_test.go

```diff
@@ -0,0 +1,325 @@
+package media
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestProtectionMetadata_MarshalMeta(t *testing.T) {
+
+	meta := &Meta{
+		EncScheme:   2,
+		IsProtected: true,
+		Offset:      2,
+		Kid:         [16]byte{42, 42, 107, 105, 100, 69, 120, 97, 109, 112, 108, 101, 58, 41, 42, 42},
+		Iv:          [16]byte{42, 42, 105, 118, 69, 120, 97, 109, 112, 108, 101, 58, 41, 42, 42},
+	}
+
+	bytes := meta.Marshal(nil, nil)
+	require.Equal(t, []byte{40, 2, 42, 42, 107, 105, 100, 69, 120, 97, 109, 112, 108, 101, 58, 41, 42, 42, 42, 42, 105, 118, 69, 120, 97, 109, 112, 108, 101, 58, 41, 42, 42, 0}, bytes)
+
+	_meta := &Meta{}
+	require.Nil(t, _meta.Unmarshal(bytes), "failed to unmarshal meta")
+
+	verifyMeta(t, meta, _meta)
+}
+
+func TestProtectionMetadata_MarshalSubsamples(t *testing.T) {
+
+	subsamples := &Subsamples{
+		SubsampleCount:  3,
+		SubsamplesTotal: 90,
+		Subsamples: []Subsample{
+			{
+				ClearDataByteSize:     10,
+				ProtectedDataByteSize: 20,
+			}, {
+				ClearDataByteSize:     10,
+				ProtectedDataByteSize: 20,
+			}, {
+				ClearDataByteSize:     10,
+				ProtectedDataByteSize: 20,
+			},
+		},
+	}
+
+	bytes := subsamples.Marshal()
+	require.Equal(t, []byte{3, 0, 10, 0, 0, 0, 20, 0, 10, 0, 0, 0, 20, 0, 10, 0, 0, 0, 20}, bytes)
```

Comment:
You can't know if the sub-samples pairs were marshalled in the correct order... Use different values...